riffusion/streamlit/pages/audio_to_audio_batch.py

```diff
@@ -0,0 +1,221 @@
+from random import randint
+
+
+def render_audio_to_audio_batch() -> None:
+    st.set_page_config(layout="wide", page_icon="🎸")
+
+    st.subheader(":wave: Audio to Audio Batch")
+    st.write(
+        """
+    Generate audio in batches from audio file and text prompt.
+    """
+    )
+
+    with st.expander("Help", False):
+        st.write(
+            """
+            This tool allows you to upload an audio file of arbitrary length and modify it with
+            a text prompt. It does this by slicing the source audio to to a spectrogram, then 
+            performing an img2img style transfer with riffusion in randomly seeded batches.
+
+            Try a denoising strength of 0.4 for light modification and 0.55 for more heavy
+            modification. The best specific denoising depends on how different the prompt is
+            from the source audio. You can play with the seed to get infinite variations.
+            """
+        )
+
+    device = streamlit_util.select_device(st.sidebar)
+
+    audio_file = st.file_uploader(
+        "Upload audio",
+        type=["mp3", "m4a", "ogg", "wav", "flac"],
+        label_visibility="collapsed",
+    )
+
+    if not audio_file:
+        st.info("Upload audio to get started")
+        return
+
+    st.write("#### Original")
+    st.audio(audio_file)
+
+    segment = streamlit_util.load_audio_file(audio_file)
+
+    segment = segment.set_frame_rate(44100)
+    st.write(f"Duration: {segment.duration_seconds:.2f}s, Sample Rate: {segment.frame_rate}Hz")
+
+    if "counter" not in st.session_state:
+        st.session_state.counter = 0
+
+    def increment_counter():
+        st.session_state.counter += 1
+
+    cols = st.columns(3)
+    clip_start_time_s = cols[0].number_input(
+        "Start Time [s]",
+        min_value=0.0,
+        value=0.0,
+    )
+    clip_duration_s = cols[1].number_input(
+        "Clip Duration [s]",
+        min_value=3.0,
+        max_value=10.0,
+        value=5.0,
+    )
+    batches = int(cols[2].number_input(
+        "Batches",
+        min_value=1,
+        max_value=100,
+        value=1,
+    ))
+    st.write(
+        f"Slicing clip of duration {clip_duration_s}s "
+        f"over {batches} batches."
+    )
+
+    clip_start_time_ms = int(clip_start_time_s * 1000)
+    clip_duration_ms = int(clip_duration_s * 1000)
+    clip_segment: pydub.AudioSegment = segment[clip_start_time_ms: clip_start_time_ms + clip_duration_ms]
+
+    audio_bytes = io.BytesIO()
+    clip_segment.export(audio_bytes, format="wav")
+
+    params = SpectrogramParams()
+
+    init_image: Image.Image = streamlit_util.spectrogram_image_from_audio(
+        clip_segment,
+        params=params,
+        device=device,
+    )
+
+    closest_width = int(np.ceil(init_image.width / 32) * 32)
+    closest_height = int(np.ceil(init_image.height / 32) * 32)
+    init_image_resized = init_image.resize((closest_width, closest_height), Image.BICUBIC)
+
+    st.write("#### Source Clip")
```

Comment:
Allows users to preview the slice they have configured in first options set. However as the spectrogram is quite large vertically, have added this in an optional `expander`.